### ISSUE TITLE
can't upload picture via immich-cli: "id must be a UUID"

### ISSUE THREAD

**Issue by teto** [immich-app/immich#13236]
### The bug

Since immich just got supported by the nix package manager, I first contacted my linux distribution to see if the problem could stem from there or from my nginx config: https://github.com/NixOS/nixpkgs/issues/346275

I can upload pictures from the website on my immich server but doing it from CLI doesn't work:
`HttpException(400): {"message":["id must be a UUID"],"error":"Bad Request","statusCode":400}`

### The OS that Immich Server is running on

nixos-unstable ~ october 2024

### Version of Immich Server

v1.117.0

### Version of Immich Mobile App

Not used

### Version of Immich CLI

2.2.22

### Platform with the issue

- [X] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
none
```

### Your .env content

```Shell
machine learning disabled
```

### Reproduction steps

I am trying to upload 
```
$ immich upload ~/Photos/toto.jpg
Crawling for assets...
Checking files | ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ | 0% | ETA: 0s | 0/1 assets
Found 0 new files and 0 duplicates
Failed to verify 1 file:
- /home/teto/Photos/toto.jpg - Error: Error: 400
All assets were already uploaded, nothing to do.
```

The last sentence `All assets were already uploaded, nothing to do` seems wrong: nothing got uploaded.

### Relevant log output

I've had the same error with immich 1.115, 1.116 and now 1.117 on the server logs: 
```shell

HttpException(400): {"message":["id must be a UUID"],"error":"Bad Request","statusCode":400}
```

### Additional information

Unrelated but using `--skip-hash` I got:
```
Crawling for assets...
Skipping hash check, assuming all files are new
Uploading assets | ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ | 0% | ETA: 0s | 0/57.6 kB
Successfully uploaded 0 new assets (0 B)
Failed to upload 1 asset:
- /home/teto/photo.jpg - TypeError: fetch failed
```

immich is beautiful. You are doing an amazing work that might finally make me happy. I am using nextcloud memories at the moment to handle my photos but immich looks like the software I 've been waiting !

**Comment by alextran1502** [comment]
Hello, what is the CLI version?

**Comment by teto** [comment]
oups, I set the CLI version in the `Version of Immich Mobile App` but I dont use the mobile app (yet). To answer your question, I am using cli 2.2.22

**Comment by C-Otto** [comment]
Immich v.1.117.0 comes with CLI 2.2.23, maybe that's the issue?

**Comment by teto** [comment]
I had the same error with 2.2.23 and then I created a new API key for another computer and it worked straightaway. I compared the config with my first computer and I had different immich uris: one had `https://` the other just `http` (I have nginx redirect for https). Changing the uri to use https fixed the whole issue. The error message looks misleading but I am not experienced with nginx to push this forward.
